File: csharp/demo/demo07-terminate.cs
/*
 * FORCING A THREAD TO TERMINATE (i.e. killing the thread)
 * Using a flag to notify the thread
 *
 * Note: The "volatile" keyword is explained in another demo.
 */
using System;
using System.Threading;



class Demo07 : IRunnable
{
    static volatile bool flagStop;

    public void run()
    {
        flagStop = false;

        var th = new Thread(() => {
            while (true)
            {
                if (flagStop)
                    break;

                Console.WriteLine("Running...");

                Thread.Sleep(1000);
            }
        });

        th.Start();

        Thread.Sleep(3000);

        flagStop = true;
    }
}

File: python/demo07_terminate.py
'''
FORCING A THREAD TO TERMINATE (i.e. killing the thread)
Using a flag to notify the thread
'''

import time
import threading



flag_stop = False



def do_task():
    while True:
        if flag_stop:
            break

        print('Running...')
        time.sleep(1)



th = threading.Thread(target=do_task)
th.start()

time.sleep(3)
flag_stop = True
